.claude/commands/console-assist.md

```diff
@@ -0,0 +1,110 @@
+# Console Assist - Package Update
+
+Expert package update assistant for OpenShift Console. Update packages safely with automated testing, building, and fixing.
+
+## Command
+`/console-assist update <package>`
+
+## Mission
+Update the specified package to its latest compatible version, ensuring builds and tests pass through automated fixing of breaking changes.
+
+## Process
+
+### Phase 1: Analysis & Update
+1. **Identify**: Read `package.json` for current version (dependencies/devDependencies/resolutions)
+2. **Research**: Use `yarn info <package> version` for latest version
+3. **Interactive Gather**: Ask user:
+   - "Do you have a migration guide URL for this update?" (optional)
+   - "Do you have a changelog/release notes URL?" (optional)
+   - "Any specific breaking changes you're aware of?" (optional)
+   - If provided, fetch and analyze these resources using WebFetch
+4. **Update**: Modify `package.json` and resolutions; run `yarn install`; report peer dependency warnings
+
+### Phase 2: Build
+5. **Build**: Run `yarn build`; capture all errors with file paths and line numbers
+6. **Analyze**: On success → Phase 3. On failure → categorize errors (TypeScript/import/API) → Phase 4
+
+### Phase 3: Test
+7. **Test**: Run `yarn test`; capture failures, deprecations, stack traces
+8. **Analyze**: On success → Phase 5. On failure → identify API changes → Phase 4
+
+### Phase 4: Auto-Fix
+9. **Fix Build Errors**:
+   - For each error, read file and identify issue (import/type/API change)
+   - Use migration guide/changelog (if provided) to inform fixes
+   - Apply fixes: update imports, types, method calls
+   - If stuck, ask user: "I'm seeing error X in file Y. The migration guide doesn't cover this. Do you know the fix or have additional documentation?"
+10. **Fix Tests**: Read test files, update expectations/mocks for API changes
+11. **Verify**: Re-run build → tests. Repeat Phase 4 if new errors. Max 3 iterations (then ask user for guidance)
```

Comment:
Does this mean 3 total iterations or 3 per error category?

Scene 1: 3 Total Fix Cycles
  Iteration 1: Fix all errors → run build → run tests → check
  Iteration 2: Fix new errors → run build → run tests → check
  Iteration 3: Fix new errors → run build → run tests → check
  → After 3 full cycles, ask user

Scene 2: 3 Attempts Per Phase
  Build phase: Try fixing build errors up to 3 times
  Test phase: Try fixing test errors up to 3 times
  → Potentially 6 total cycles

Scene 3: 3 Attempts Per File/Error
  For each file with errors, retry up to 3 times
  → Could be many more than 3 total iterations

or more scenes?